src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,478 @@
+                }
+
+                // Check if the current item has a sliding expiration and is going to expire within the
+                // next timeout period(ex:20 minutes), then extend the expiration time.
+                if (slidingExpiration.HasValue &&
+                    utcNow >= (oldExpirationTime - slidingExpiration.Value))
+                {
+                    var newExpirationTime = GetExpirationTime(slidingExpiration, absoluteExpiration, utcNow);
+
+                    if (oldExpirationTime != newExpirationTime)
+                    {
+                        reader.Dispose();
+
+                        command = new SqlCommand(_sqlQueries.UpdateCacheItemExpiration, connection);
+                        command.Parameters.AddWithValue(CacheItemId, key);
+                        command.Parameters.AddWithValue(ExpiresAtTimeUTC, newExpirationTime);
+
+                        await command.ExecuteNonQueryAsync();
+                    }
+                }
+            }
+
+            StartScanForExpiredCacheItems();
+
+            return cacheItemvalue;
+        }
+
+        public void Refresh([NotNull] string key)
+        {
+            Get(key);
+        }
+
+        public async Task RefreshAsync([NotNull] string key)
+        {
+            await GetAsync(key);
+        }
+
+        public void Remove([NotNull] string key)
+        {
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.DeleteCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
+
+                connection.Open();
+
+                command.ExecuteNonQuery();
+            }
+
+            StartScanForExpiredCacheItems();
+        }
+
+        public async Task RemoveAsync([NotNull] string key)
+        {
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.DeleteCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
+
+                await connection.OpenAsync();
+
+                await command.ExecuteNonQueryAsync();
+            }
+
+            StartScanForExpiredCacheItems();
+        }
+
+        public void Set([NotNull] string key, [NotNull] byte[] value, [NotNull] DistributedCacheEntryOptions options)
+        {
+            var utcNow = _systemClock.UtcNow;
+            var expirationInfo = GetExpirationInfo(utcNow, options);
+
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                // Try updating the cache item and if it does not succeed then its probably a new item
+                var command = new SqlCommand(_sqlQueries.UpdateCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
+                command.Parameters.AddWithValue(CacheItemValue, value);
```

Comment:
Consider setting the size of the value parameter as well to avoid query cache fragmentation on the server (feel free to double-check that is has an incidence before you do it). What you can do here is specify an arbitrary size in which the most typical payloads will fit unless the payload is larger than that size.